src/schema/users.ts

```diff
@@ -2364,6 +2377,43 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         .update({ id: ctx.userId }, { cover: imageUrl });
       return getCurrentUser(ctx, info);
     },
+    uploadResume: async (
+      _,
+      { resume }: { resume: Promise<FileUpload> },
+      ctx: AuthContext,
+    ): Promise<GQLEmptyResponse> => {
+      if (!resume) {
+        throw new ValidationError('File is missing!');
+      }
+
+      const upload = await resume;
+
+      // Validate file extension
+      const extension = upload.filename?.split('.')?.pop()?.toLowerCase();
+      if (extension !== 'pdf') {
+        throw new ValidationError('Extension must be .pdf');
+      }
+
+      // Buffer the stream
+      const chunks: Buffer[] = [];
+      const stream = upload.createReadStream();
+      for await (const chunk of stream) {
+        chunks.push(chunk);
+      }
+      const buffer = Buffer.concat(chunks);
+
+      // Validate MIME type using buffer
+      const detectedFileType = getMimeTypeFromArrayBuffer(buffer);
+      if (detectedFileType !== 'application/pdf') {
+        throw new ValidationError('File is not a PDF');
+      }
+
+      // Actual upload using buffer as a stream
+      const filename = `${ctx.userId}.pdf`;
+      await uploadResumeFromStream(filename, Readable.from(buffer));
```

Comment:
is it a little too much of juggling between different stream/buffer/readable? Seems like we should not need to juggle around types that much.